src/dev/create-extension.ts

```diff
@@ -77,13 +103,47 @@ const findTemplateFiles = async (dir: string, templates: Set<string>) => {
   }
 };
 
-const copyFiles = async (files: string[], projectName: string, projectPath: string, templates: Set<string>) => {
-  for (const file of files) {
+const copyFiles = async (
+  changedFiles: string[],
+  deletedFiles: string[],
+  projectName: string,
+  projectPath: string,
+  templates: Set<string>,
+) => {
+  for (const file of deletedFiles) {
+    const destPath = path.join(EXTERNAL_EXTENSIONS_DIR, projectName, TARGET_EXTENSION_DIR, file);
+    if (fs.existsSync(destPath)) {
+      await fs.promises.unlink(destPath);
+      prettyLog.success(`Removed deleted file: ${file}`, 2);
+      console.log("\n");
+
+      // remove empty directories
+      const dirPath = path.dirname(destPath);
+      try {
+        const remainingFiles = await fs.promises.readdir(dirPath);
+        if (remainingFiles.length === 0) {
+          await fs.promises.rmdir(dirPath);
+          prettyLog.success(`Removed empty directory: ${path.relative(EXTERNAL_EXTENSIONS_DIR, dirPath)}`, 2);
+          console.log("\n");
+        }
+      } catch {
+        // directory might already be deleted, ignore error
```

Comment:
How might it already be deleted if file existed before deleting? Sometimes folder is removed automatically?

Reply:
Actually while testing I by mistake ran the `yarn create-extension ../ctf-extension-instance` in two terminals at same time and it broke, lol I mean very rare case but its good to handle this?